src/transformers/models/mamba2/convert_mamba2_ssm_checkpoint_to_pytorch.py

```diff
@@ -15,31 +15,94 @@
+
+    # Padded vocab size, mostly of 16 but 32 is also very common in different models
+    vocab_size = config_ssm["vocab_size"]
+    pad_vocab_size_multiple = config_ssm["pad_vocab_size_multiple"]
+    if (vocab_size % pad_vocab_size_multiple) != 0:
+        vocab_size += pad_vocab_size_multiple - (vocab_size % pad_vocab_size_multiple)
+    hf_config.vocab_size = vocab_size
+
+    return hf_config, is_not_codestral
+
+
+def load_state_dict_from_safetensors(mamba2_checkpoint_path: str) -> Dict[str, torch.Tensor]:
     # Load weights and config from paths
     original_state_dict = {}
     with safe_open(mamba2_checkpoint_path, framework="pt") as f:
         for k in f.keys():
             newk = k.removeprefix("model.")
             original_state_dict[newk] = f.get_tensor(k).clone()
+    return original_state_dict
+
+
+def load_state_dict_from_torch(mamba2_checkpoint_path: str) -> Dict[str, torch.Tensor]:
+    return torch.load(mamba2_checkpoint_path + "/pytorch_model.bin", map_location="cpu")
+
 
+def convert_mamba2_checkpoint_file_to_huggingface_model_file(
+    mamba2_checkpoint_path: str, precision: str, output_dir: str, tokenizer_model_path: Optional[str] = None
+) -> None:
+    # Load and save config based on name
+    config_path = mamba2_checkpoint_path
+    config_path = (
+        config_path + "/params.json" if path.isfile(config_path + "/params.json") else config_path + "/config.json"
+    )
+    with open(config_path, "r", encoding="utf-8") as json_file:
+        config = json.load(json_file)
+    hf_config, is_not_codestral = convert_ssm_config_to_hf_config(config)
+    hf_config.save_pretrained(output_dir)
+
+    # Load state dict of the original model
+    state_dict_load_function = load_state_dict_from_torch if is_not_codestral else load_state_dict_from_safetensors
+    original_state_dict = state_dict_load_function(mamba2_checkpoint_path)
+
+    # Load and transfer to hf model
+    hf_model = Mamba2ForCausalLM(hf_config)
     hf_model.load_state_dict(original_state_dict)
 
     # Save new model to pytorch_dump_path
-    hf_model.to(torch.bfloat16).save_pretrained(output_dir)
-    tokenizer_class = LlamaTokenizerFast
-    tokenizer = tokenizer_class(tokenizer_model_path, legacy=False, from_slow=True)
+    dtype = torch.float32 if precision == "fp32" else (torch.bfloat16 if precision == "bf16" else torch.float16)
+    save_model(hf_model.to(dtype), output_dir + "/model.safetensors", metadata={"format": "pt"})
+
+    # Load and save tokenizer
+    if tokenizer_model_path is not None and not is_not_codestral:
+        tokenizer_class = LlamaTokenizerFast
+        tokenizer = tokenizer_class(tokenizer_model_path, legacy=False, from_slow=True)
+    else:
+        tokenizer = AutoTokenizer.from_pretrained("state-spaces/mamba-130m-hf")
+        # TODO: how to correctly overwrite padding side so it is saved as is
+        tokenizer.padding_side = "left"
     tokenizer.save_pretrained(output_dir)
```

Comment:
Don't know how to overwrite the padding side. Would appreciate some help here. Mamba uses the same tokenizer for all models so this should suffice.

Reply:
Since we're using `AutoTokenizer`, I'm not sure it's possible as the parameters from this tokenizer family would be used. Llama family is left-side, but AutoTokenizer mapping will give GPTNeoX family which is right-side, so indeed saving and loading will not cut it - an alternative would be to create a `Mamba2Tokenizer` class perhaps?

Reply:
I've added a `Mamba2Tokenizer` which only overrides the padding side and keeps everything else from the original `GPTNeoXTokenizer`. 

There are warnings though because I overloaded to a different class. Should I leave the warnings? And if not, is there an easy way to suppress them temporarily?